src/layout.rs

```diff
@@ -67,26 +68,49 @@ impl Event {
     }
 }
 
+/// The various states used internally by Keyberon
 #[derive(Debug, Clone, Copy, Eq, PartialEq)]
 enum State {
-    NormalKey { keycode: KeyCode, coord: (u8, u8) },
-    LayerModifier { value: usize, coord: (u8, u8) },
+    /// Normal keys
```

Comment:
That's an internal structure, not exposed to the world, so no need to doccomment (and these doc comments don't really help)

Reply:
Well, you said you wanted to document everything :smile:

I actually added them because Rust was complaining that not everything had a docstring...  So I went to town adding them to everything just to keep it happy.  Though some of them seem important/make sense *to me*.  For example:

```rust
    /// Fake key event for sequences
    FakeKey {
        keycode: KeyCode,
    },
```

I wouldn't want to remove that one because the point of a `FakeKey` isn't obvious.  If I were looking at this code for the first time I'd think, "Why TF is that there?" and it wouldn't immediately occur to me that it's something that's necessary for sequences/macros.

Should it just be a comment?  I'll remove the others.

Reply:
You can of course put interesting comments. In this case, better something like "A key generated by some event. Mainly used by SeqenceEvent".